lib/Demangling/Demangler.cpp

```diff
@@ -93,8 +93,12 @@ static bool isRequirement(Node::Kind kind) {
 
 void swift::Demangle::failAssert(const char *file, unsigned line,
                                  NodePointer node, const char *expr) {
-  fprintf(stderr, "%s:%u: assertion failed for Node %p: %s", file, line, node,
+  fprintf(stderr, "%s:%u: assertion failed for Node %p: %s\n", file, line, node,
```

Comment:
We should use `swift_reportError` and `fatalError` to get these messages into the crash log too.

Reply:
Hmmm. That might be slightly trickier, in that `node->dump()` isn't going to send its output to the right place. It's not a bad idea to get something in there though; I'll have a look how that all works.

Reply:
It isn't quite as simple as I'd like; we can't call `swift_reportError` or `fatalError` without linking `swiftRuntime`, which not everything that includes the demangler code does.